Towny/src/main/java/com/palmergames/bukkit/towny/object/Nation.java

```diff
@@ -602,43 +604,58 @@ public int getNationZoneSize() {
 		if (!TownySettings.getNationZonesEnabled())
 			return 0;
 		
-		return TownySettings.getNationLevel(this).nationZonesSize();
+		return getNationLevel().nationZonesSize();
 	}
 
 	/**
-	 * Get the Nation's current Nation Level.
+	 * Get the Nation's current NationLevel.
 	 * <p>
 	 *     Note that Nation Levels are not hard-coded. They can be defined by the server administrator,
 	 *     and may be different from the default configuration.	 
 	 * </p>
-	 * @return Nation Level (int) for current population.
+	 * @return NationLevel of the nation.
 	 */
-	public int getLevel() {
-		return getLevel(this.getNumResidents());
+	public NationLevel getNationLevel() {
+		return TownySettings.getNationLevel(this);
 	}
 
 	/**
-	 * Get the maximum level a Nation may achieve.
-	 * @return Size of TownySettings' configNationLevel SortedMap.
+	 * Get the Nation's NationLevel assuming a supplied modifier. This is used to
+	 * determine theoretical or future nationLevels given a change in the resident
+	 * amount or town amount.
+	 * <p>
+	 * Note that Nation Levels are not hard-coded. They can be defined by the server
+	 * administrator, and may be different from the default configuration.
+	 * </p>
+	 * 
+	 * @param modifier an int representing either the amount of residents or towns
+	 *                 that determines a nation's NationLevel.
+	 *                 {@link TownySettings#isNationLevelDeterminedByTownCount()}
+	 *                 will tell you which is used.
+	 * @return NationLevel of the nation knowing the modifier.
 	 */
-	public int getMaxLevel() {
-		return TownySettings.getConfigNationLevel().size();
+	public NationLevel getNationLevelWithModifier(int modifier) {
+		return TownySettings.getNationLevelWithModifier(modifier);
 	}
```

Comment:
This method isn't really needed here, since it only delegates to TownySettings and doesn't do anything with this nation instance

Reply:
I think I was using this in an earlier iteration of this branch.